src/main/java/com/uber/cadence/internal/sync/WorkflowThreadImpl.java

```diff
@@ -236,31 +236,25 @@ public void start() {
         .gauge(MetricsType.WORKFLOW_ACTIVE_THREAD_COUNT)
         .update(((ThreadPoolExecutor) threadPool).getActiveCount());
 
-    try {
-      taskFuture = threadPool.submit(task);
-      return;
-    } catch (RejectedExecutionException e) {
-      getDecisionContext()
-          .getMetricsScope()
-          .counter(MetricsType.STICKY_CACHE_THREAD_FORCED_EVICTION)
-          .inc(1);
+    while (true) {
```

Comment:
Should we have some time limit, just to be on the safe side?

Reply:
We break if there's no item in the cache that can be evicted. Seems to me it's safe to always retry if we have a successful eviction.